### DISCUSSION TITLE
Heat extraction in a doublet

### DISCUSSION THREAD

**Discussion by Josselin-76** [idaholab/moose#24878] [Q&A General] [answered]
Dear community,

I am modeling a doublet in a monolayer reservoir only in TH.
![image](https://github.com/idaholab/moose/assets/93592660/dce515c9-6a24-4e45-b577-b52838a1a013)

My injection work perfectly however my extraction does not work. The temperature a the production well is heavily increasing over time instead of remaining constant and then cooling.

I have used all the available approaches in Point and line sources/sinks. 
1) Injection in DirichletBC : imposing water pressure and temperature at the well imposing water pressure at the production well

![DirichletBC](https://github.com/idaholab/moose/assets/93592660/3c6608a8-868d-4a49-b5be-5257ce323a7e)

![image](https://github.com/idaholab/moose/assets/93592660/e75c5ca9-e20e-47a9-b1e5-29272463747c)

2) Injection with [PorousFlowPointEnthalpySourceFromPostprocessor] and extraction with [PorousFlowPointSourceFromPostprocessor]

![image](https://github.com/idaholab/moose/assets/93592660/847bee4a-3338-4bde-89ea-289a59dc1f03)

3)  Injection with [PorousFlowPointSourceFromPostprocessor] and extraction with [PorousFlowPeacemanBorehole]
![image](https://github.com/idaholab/moose/assets/93592660/f0612390-9808-45c1-bbd2-b01745354315)

4) Injection with water pressure PorousFlowSquarePulsePointSource, temperature Dirichlet BC, extraction PorousFlowSquarePulsePointSource

![image](https://github.com/idaholab/moose/assets/93592660/a99ed1e2-60ed-4f0a-9262-9a45e66a8bba)

5)  Injection with [PorousFlowPointSourceFromPostprocessor]  and production with PorousFlowSquarePulsePointSource

![image](https://github.com/idaholab/moose/assets/93592660/8549e0a2-e7bb-4520-8020-6aca787619ec)

Can you please help with that, in an ideal world I would like to inject with [PorousFlowPointEnthalpySourceFromPostprocessor and extract with PorousFlowSquarePulsePointSource.

Thank you

 Here is my code
[Aquifer_TH.txt](https://github.com/idaholab/moose/files/11914801/Aquifer_TH.txt)
 
Mesh 
[mesh.txt](https://github.com/idaholab/moose/files/11914811/mesh.txt)

**Comment by GiudGiud** [comment]
Hello

Disclaimer: not a subject matter expert

So the extraction is a point source (sink).
Is the mesh resolved enough near the extraction point?

@cpgr (a SME) 

Guillaume

**Reply by Josselin-76** [reply]
Thanks for the answer.

It is not mesh/node dependent. When refining and increasing the size of the refined zone, the extraction temperature still increases.

**Reply by cpgr** [reply]
You need to make sure that you are producing both mass and heat, so need two production sources/wells - one for mass, one for heat. The `PorousFlowSquarePulsePointSource` doesn't have an option to easily do enthalpy (probably something I should add). You might be able to use the two 'source from post processor' point sources as sinks (just use a -ve mass flux).

I think the Paceman borehole approach should have worked though?

**Reply by cpgr** [reply, marked as answer]
Perhaps like
```
[DirakKernels]
  [fluid_source]
    type = PorousFlowPointSourceFromPostprocessor
    variable = pwater
    mass_flux = mass_flux_in
    point = '5150 5000 500'
  []
  [enthalpy_source]
    type = PorousFlowPointEnthalpySourceFromPostprocessor
    variable = temp
    mass_flux = mass_flux_in
    point = '5150 5000 500'
    T_in = T_in
    pressure = pwater
    fp = water
  []
  [fluid_sink]
    type = PorousFlowPointSourceFromPostprocessor
    point = '4850 5000 500'
    variable = pwater
    mass_flux = mass_flux_out #[kg/s]
  []
  [enthalpy_sink]
    type = PorousFlowPointEnthalpySourceFromPostprocessor
    variable = temp
    mass_flux = mass_flux_out
    point = '4850 5000 500'
    T_in = temp_pro
    pressure = pwater
    fp = water
  []
[]
```
